src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -209,38 +190,35 @@ private function createRelationsForItem($item, $formattedData)
      * Those will be nested accordingly in this relation array, so address relation will have a nested relation with country.
      *
      *
-     * @param  array  $input  The form data.
-     * @return array The formatted relation data.
+     * @param  array  $input  The form input.
+     * @return array The formatted relation details.
      */
-    private function getRelationDataFromFormData($data)
+    private function getRelationDetailsFromInput($input)
     {
-        $relation_fields = $this->getRelationFields();
+        $relation_fields = $this->getRelationFieldsWithoutPivot();
 
-        //remove fields that are not in the submitted form data
-        $relation_fields = array_filter($relation_fields, function ($item) use ($data) {
-            return Arr::has($data, $item['name']);
+        //remove fields that are not in the submitted form input
+        $relation_fields = array_filter($relation_fields, function ($item) use ($input) {
+            return Arr::has($input, $item['name']);
         });
 
-        $relationData = [];
+        $relationDetails = [];
         foreach ($relation_fields as $relation_field) {
```

Comment:
Inside this `foreach` everything is a bit obscure... I can't easily tell what's happening. Can you please add comments explaining what is happening and why?

Thanks.